### PR TITLE
Created Book Detail Screen

### PR DESCRIPTION
**Pull request by MatiasMercado**
# Summary
- Created Book Detail View
- Added Routing for Dashboard and Book Detail

## Trello Card
https://trello.com/c/mBaXbn0g/17-book-detail

## Screenshot
bookdetail2

### CODE REVIEW

File: wbooks/src/screens/App/BookDisplay/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+              title={this.book.title}
+              year={this.book.year}
+              imageUrl={this.book.image_url}/>
+            <Suggestions/>
+            <Comments/>
+          </div>
+        </div>
+      </div> :
+      <div className='not-found'>
+        <h2>Este no es el libro que estas buscando</h2>
```
Comment (ignaciosantise) [inline wbooks/src/screens/App/BookDisplay/index.js]: move this string to a strings file

File: wbooks/src/shared/Book/index.js [hunk 2]
```diff
@@ -1,24 +1,17 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import strings from './strings.js'
-import './style.css'
+import { Link } from 'react-router-dom';
+import BookImage from 'shared/BookImage';
+import strings from 'shared/strings.js'
```
Comment (ignaciosantise) [inline wbooks/src/shared/Book/index.js]: why did you created a single string file? 👼  its good to have them separated! 😄  maybe you could extract common strings to this file

File: wbooks/src/shared/BookDetail/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          <h4 className='book-info-item'>{this.props.author}</h4>
+          <h4 className='book-info-item'>{this.props.year}</h4>
+          <h4 className='book-info-item'>{this.props.genre}</h4>
+          <h4 className='book-info-description'>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
+            incididunt ut labore et dolore magna aliqua.
+            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
+            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
+            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+          </h4>
+          <input className='rent-button' type='submit' value='Alquilar'/>
```
Comment (ignaciosantise) [inline wbooks/src/shared/BookDetail/index.js]: Extract 'Alquilar' to a string file

File: wbooks/src/shared/BookDetail/index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  render() {
+    return (
+      <div className='book-detail'>
+        <BookImage imageStyle='book-detail-image' imageUrl={this.props.imageUrl}/>
+        <div className='book-info'>
+          <h3 className='book-info-title'>{this.props.title}</h3>
+          <h4 className='book-info-item'>{this.props.author}</h4>
+          <h4 className='book-info-item'>{this.props.year}</h4>
+          <h4 className='book-info-item'>{this.props.genre}</h4>
+          <h4 className='book-info-description'>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
```
Comment (ignaciosantise) [inline wbooks/src/shared/BookDetail/index.js]: book `description` should come from a prop, like `title`,`author`...

File: wbooks/src/shared/BookImage/index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import strings from 'shared/strings.js';
+import './style.css';
+
+export default (props) => {
+  // imageStyle must be the name of a CSS Class
+  return  props.imageUrl ?
+      <img src={props.imageUrl} alt='cover' className={'book-image ' + props.imageStyle}/> :
+      <div className={'book-image-default ' + props.imageStyle}>
```
Comment (ignaciosantise) [inline wbooks/src/shared/BookImage/index.js]: Check this! http://es6-features.org/#StringInterpolation

you could do sth like this:
```
className={`book-image-default` ${props.imageStyle}}
```

File: wbooks/src/shared/CommentsSection/index.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+export default () => {
+  return (
+    <div className='comments-section'>
+      <h2 className='comments-section-title'>Comentarios</h2>
+
+      <div className='comment-block'>
+        <img className='user-icon' src='images/default-user.png' alt='default_user'/>
+        <form className='add-comment'>
+          <h3 className='comment-block-title'>Agregar comentario</h3>
```
Comment (ignaciosantise) [inline wbooks/src/shared/CommentsSection/index.js]: string file 😄

File: wbooks/src/shared/CommentsSection/index.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+          <textarea className='comment-text' rows='4' cols='80'/>
+          <input className='send-button' type='submit' value='Enviar'/>
```
Comment (ignaciosantise) [inline wbooks/src/shared/CommentsSection/index.js]: string file :D

File: wbooks/src/shared/BookDetail/style.css [hunk 8]
```diff
@@ -0,0 +1,49 @@
+.book-detail {
+  border-bottom: 1px solid #00adee;
+  display: flex;
+  padding: 50px 0px;
+  width: 100%;
+}
+
+.book-detail-image {
+    /*height: 100%;*/
```
Comment (ignaciosantise) [inline wbooks/src/shared/BookDetail/style.css]: remove this comment

File: wbooks/src/shared/CommentsSection/index.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      <div className='comment-block'>
+        <img className='user-icon' src='images/default-user.png' alt='default_user'/>
+        <div className='add-comment'>
+          <h3 className='comment-block-title'>Kimberly Carter</h3>
+          <h4 className='comment-date'>xx/xx/xx</h4>
+          <div className='comment-text'>Lorem ipsum dolor sit amet, consectetur adipiscing elit,
+            sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
+          </div>
+          <input className='send-button' type='submit' value='Enviar'/>
```
Comment (ignaciosantise) [inline wbooks/src/shared/CommentsSection/index.js]: do you need the send button below the comments? or just below the input field?

File: wbooks/src/shared/strings.js [hunk 10]
```diff
@@ -1,6 +1,13 @@
...
+  // Filters
   authorFilterName: 'author',
-  titleFilterName: 'title'
+  titleFilterName: 'title',
+  // Default Values
+  defaultAuthor: 'Autor',
+  defaultImageUrl: 'images/default_book.svg',
+  defautlGenre: 'Temática',
+  defautlYear: 'Año',
+  defautlTitle: 'Titulo'
```
Comment (ignaciosantise) [inline wbooks/src/shared/strings.js]: `Título`

File: wbooks/src/screens/App/BookDisplay/index.js (hunk 1 above)
Comment (MatiasMercado) [reply wbooks/src/screens/App/BookDisplay/index.js]: Done

File: wbooks/src/shared/Book/index.js (hunk 2 above)
Comment (MatiasMercado) [reply wbooks/src/shared/Book/index.js]: Done

File: wbooks/src/shared/BookDetail/index.js (hunk 4 above)
Comment (MatiasMercado) [reply wbooks/src/shared/BookDetail/index.js]: Done

File: wbooks/src/shared/BookDetail/index.js (hunk 3 above)
Comment (MatiasMercado) [reply wbooks/src/shared/BookDetail/index.js]: Done

File: wbooks/src/shared/BookImage/index.js (hunk 5 above)
Comment (MatiasMercado) [reply wbooks/src/shared/BookImage/index.js]: Checked!

File: wbooks/src/shared/CommentsSection/index.js (hunk 6 above)
Comment (MatiasMercado) [reply wbooks/src/shared/CommentsSection/index.js]: Done

File: wbooks/src/shared/CommentsSection/index.js (hunk 7 above)
Comment (MatiasMercado) [reply wbooks/src/shared/CommentsSection/index.js]: Done

File: wbooks/src/shared/CommentsSection/index.js (hunk 9 above)
Comment (MatiasMercado) [reply wbooks/src/shared/CommentsSection/index.js]: You are right, they are not needed

File: wbooks/src/shared/BookDetail/style.css (hunk 8 above)
Comment (MatiasMercado) [reply wbooks/src/shared/BookDetail/style.css]: Done

File: wbooks/src/shared/strings.js (hunk 10 above)
Comment (MatiasMercado) [reply wbooks/src/shared/strings.js]: Done

**Comment by ignaciosantise** [comment]
Try to be more descriptive in the commit messages, and its good to separate fixes in different commits 😄

**Pull request merged**